src/ad_extension.js

```diff
@@ -1,16 +1,13 @@
-export class AdExtension {
-  constructor() {
-    this.name = null;
-    this.value = null;
-    this.attributes = {};
-    this.children = [];
-  }
-
-  isEmpty() {
+export function AdExtension() {
+  this.name = null;
+  this.value = null;
+  this.attributes = {};
+  this.children = [];
+  this.isEmpty = function() {
     return (
```

Comment:
this should be moved out to a new function

Reply:
Not sure how you would want me to do it. Changing all except AdExtension.

Reply:
I mean that in this file you could have two functions: one to create the object, and another that checks if the extension object is empty (by receiving the object as a param)